adodb.inc.php

```diff
@@ -5585,7 +5631,15 @@ function ADONewConnection($db='') {
 				return false;
 			}
 
-			$cls = 'ADODB_'.$db;
+			if ($pdoExtension)
+			{
+				$cls = 'ADODB_pdo_' . $pdoExtension;
+				$db = 'PDO';
+			}
+			else
+			{
+				$cls = 'ADODB_'.$db;
+			}
```

Comment:
Also, what about performing early detection of legacy ADONewConnection('pdo') calls (i.e. when `$db == 'pdo' && $pdoExtension == ''`), either here or maybe even better when we split and process the pdo driver string above, and throw an error instead of delaying this until ADODB_pdo::_connect()